### PR TITLE
Jcs/crop lookup

### PR DESCRIPTION
**Pull request by jSylvestre**
This has the DB part of the issue if someone else wants to do the typeahead change
Part of #523

### CODE REVIEW

File: Harvest.Web/Controllers/Api/LookupController.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        }
+
+
+        public async Task<ActionResult> SearchCrops(string type, string query)
+        {
+            //I think this is right, didn't test
+            var crops = await _dbContext.CropLookups
+                .Where(a => a.Type == type &&
+                            (EF.Functions.Like(a.Crop, query.EfStartsWith()) ||
+                             EF.Functions.Like(a.Crop, query.EfContains())))
```
Comment (sprucely) [inline Harvest.Web/Controllers/Api/LookupController.cs:28]: I think `EfStartsWith()` is a subset of `EfContains()` and so is redundant here.

**Review by sprucely — APPROVED** [review]
Looks good. Maybe just remove that unnecessary `EfStartsWith`

**Review by srkirkland — CHANGES_REQUESTED** [review]
first is I hate to have you redo the migrations but I think the classname here should be `Crop` and the table thus `Crops`.  We do lookups all the time in our DBs but don't usually use the `Lookups` naming scheme and I'm not a fan of that specificity. 

second is more simple and doesn't really need to be fixed, but I think it's unlikely we'll ever bother searching for crops.  A simple controller method to just return all crops would probably be just fine, the list will not be that long, and it'll make searching much easier via typeahead.

File: Harvest.Web/Controllers/Api/LookupController.cs (hunk 1 above)
Comment (jSylvestre) [reply Harvest.Web/Controllers/Api/LookupController.cs:28]: @sprucely I based it off what we do in Peaks

**Review by srkirkland — APPROVED** [review]

**Pull request merged**